test/checkImageSize.js

```diff
@@ -0,0 +1,86 @@
+const sizeOf = require('image-size');
+const fs = require('fs');
+const { exit } = require("process");
+
+function getFileSizeMB(fileStats){
+    return (fileStats.size) / (1024*1024);
+}
+
+//Check if image is larger than 1MB
+function isFileSizeBig(fileStats){ 
+    return getFileSizeMB(fileStats) > 1;
+}
+//Check if the width of image is larger than 1024px or lower than 512px
+function isImageWidthUnreasonable(imageSize){
+    let width = imageSize.width;
+    return (width > 1024 | width < 512);
+}
+
+function isImageWidthLarge(imageSize){
+    let width = imageSize.width;
+    return (width >1024);
+}
+
+function isImageWidthSmall(imageSize){
+    let width = imageSize.width;
+    return (width <512);
+}
+
+function isAValidExtension(imageFileName){
+    return (imageFileName.includes(".jpg") | imageFileName.includes(".png") | imageFileName.includes(".jpeg"))
+}
+
+function imageSizeChecker(input){  //input here is the image file
```

Comment:
Something I mentioned earlier is that we probably want to keep track of what types of errors are associated with each file, so we know what's fixable and what's not.

Unfixable:

* bad aspect ratio
* too small

Fixable:

* extension is incorrectly cased
* image is too big

Instead of just throwing an error, consider returning something that other functions (ex your fixer, or your logging) can use. Here, **enums** are a great tool; you can:

* define a set of possible errors (see what I noted above)
* give each element in the set some properties (the log message, and whether or not it's fixable)

Then, this function can return the enum associated with the image (and optionally, any data you need - in this case, you can return an object with an enum as one of the items). Other functions can use the enum however they wish (to log what's wrong, or run the autofixer).